### PR TITLE
Fix get node from db error

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- the trieStorageManager tried to double close the main & checkpoint storer along with the data components handler
- the node called the `Close` method for the state & data components handlers in wrong order
- activeDB marker was set even though there was a `context closing` err during snapshot
  
## Proposed changes
- fixed the closing order on 2 managed components
- removed the call to Close on storers in trieStorageManager.go
- added a dedicated component for the closing order of the managed components
- return err during snapshot if the db is closing

## Testing procedure
- standard system test - no `getNodeFromDB` errors should appear

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: trie/node.go [hunk 1]
```diff
@@ -271,14 +272,22 @@ func shouldStopIfContextDoneBlockingIfBusy(ctx context.Context, idleProvider Idl
...
 }
 
-func treatCommitSnapshotError(err error, hash []byte, missingNodesChan chan []byte) {
+func treatCommitSnapshotError(err error, hash []byte, missingNodesChan chan []byte) (bool, error) {
+	if err == nil {
+		return false, nil
+	}
+
+	if !strings.Contains(err.Error(), core.GetNodeFromDBErrorString) {
+		return false, err
```
Comment (iulianpascalau) [inline trie/node.go:281]: is this ok to return here false? I mean, the error is not of type `GetNodeFromDBErrorString` but is still an error & the node could have not been received.

**Review by raduchis — COMMENTED** [review]
looks like the code coverage is not as high as required

File: factory/mock/heartbeatV2ComponentsStub.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func (hbc *HeartbeatV2ComponentsStub) CheckSubcomponents() error {
 
 // String -
 func (hbc *HeartbeatV2ComponentsStub) String() string {
-	return ""
+	return "managedHeartbeatV2Components"
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/mock/heartbeatV2ComponentsStub.go]: was afraid of cyclic imports. However, I've changed everywhere and it looks ok.

**Review by iulianpascalau — COMMENTED** [review]
:+1: looks good from my perspective

**Review by BeniaminDrasovean — COMMENTED** [review]
:+1: from me

File: factory/mock/heartbeatV2ComponentsStub.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply factory/mock/heartbeatV2ComponentsStub.go]: ok!

**Review by raduchis — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request closed without merging**